### ISSUE TITLE
Implement configlib

### ISSUE THREAD

**Issue by ddevault** [KnightOS/KnightOS#110]
We should have a library for reading unix-style config files, such as:

```
foo=bar
thing=100
[category]
foo=baz

other_thing=10
[other_category]
# Comment
foo=foo
```

The category for the first two is "__none__". To read a config entry...

```
kld(de, configPath)
config(openConfig)
kld(hl, default_foo)
config(readOption)
kld(hl, other_category_foo)
config(readOption)
config(closeConfig)
; ...
default_foo:
    .db "foo", 0
other_category_foo:
    .db "other_category/foo", 0
```

`readOption` should allocate some memory for the requested option and copy it there.

This library should also be able to write to configs.

```
kld(de, other_category_foo)
kld(hl, some_value)
config(writeOption)
config(writeConfig)
config(closeConfig)
; ...
some_value:
    .db "some value", 0
```

tl;dr:
- [x] openConfigRead
- [x] openConfigWrite
- [x] closeConfig
- [x] readOption
- [x] readOption_8
- [x] readOption_16
- [x] readOption_32
- [ ] readOption_float
- [ ] readOption_bool
- [ ] writeOption
- [ ] writeOption_u8
- [ ] writeOption_u16
- [ ] writeOption_u32
- [ ] writeOption_s8
- [ ] writeOption_s16
- [ ] writeOption_s32
- [ ] writeOption_float
- [ ] writeOption_bool
